runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowMetrics.java

```diff
@@ -206,12 +210,21 @@ private StringSetResult getStringSetValue(MetricUpdate metricUpdate) {
     }
 
     private BoundedTrieResult getBoundedTrieValue(MetricUpdate metricUpdate) {
-      if (metricUpdate.getTrie() == null) {
+      BoundedTrieData trieData = null;
+      Object trieFromResponse = metricUpdate.getBoundedTrie();
+      // Fail-safely cast Trie returned by dataflow API to BoundedTrieResult
+      if (trieFromResponse instanceof BoundedTrie) {
+        BoundedTrie bTrie = (BoundedTrie) metricUpdate.getBoundedTrie();
+        trieData = BoundedTrieData.fromProto(bTrie);
```

Comment:
Could this just be `trieData = BoundedTrieData.fromProto((BoundedTrie) trieFromResponse);`

Reply:
Thanks for catching this. Fixed.